src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -356,13 +382,36 @@ void GinIndexStore::writeSegment()
     fst_builder.build();
     write_buf.finalize();
 
-    /// Write FST size
-    writeVarUInt(buffer.size(), *dict_file_stream);
-    current_segment.dict_start_offset += getLengthOfVarUInt(buffer.size());
+    size_t uncompressed_size = buffer.size();
+
+    /// Write uncompressed FST size
+    writeVarUInt(uncompressed_size, *dict_file_stream);
+    current_segment.dict_start_offset += getLengthOfVarUInt(uncompressed_size);
 
-    /// Write FST blob
-    dict_file_stream->write(reinterpret_cast<char *>(buffer.data()), buffer.size());
-    current_segment.dict_start_offset += buffer.size();
+    Format version = Format::v1;
```

Comment:
Here, the version field has this semantics:
- v1: The FST is unconditionally uncompressed
- v2: The FST is unconditionally compressed.

Versions are usually interpreted in a slightly different way:
- v1: The FST is unconditionally uncompressed.
- v2: The FST may be compressed or uncompressed. In other words, the v2 format has enough flexibility to accommodate for both cases; and there is another metadata bit that distinguishes between both cases.

The latter interpretation seems more desirable.